### PR TITLE
kubelet/kuberuntime: Improving test coverage

### PR DESCRIPTION
**Pull request by tmilox**
Signed-off-by: TommyStarK <thomasmilox@gmail.com>

#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

If applied this commit will increase the unit tests code coverage of `pkg/kubelet/kuberuntime`

- Initial unit tests code coverage

Screenshot 2022-12-08 at 14 48 18

Screenshot 2022-12-08 at 19 58 07

- After changes

Screenshot 2022-12-08 at 19 57 43

Screenshot 2022-12-08 at 19 56 38

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by TommyStarK** [comment]
Hello @dims, @mrunalp 

I know guys you are super busy and this PR is far from being a priority. As a very new aspiring contributor I am still trying to become familiar with the code and to start helping on small pieces. Hope you don't mind.

Any feedback will be greatly appreciated 😅.

Best regards,

**Comment by TommyStarK** [comment]
Hello @bart0sh,

Thank you for taking the time. All required pipelines passed. Job `pull-kubernetes-e2e-inplace-pod-resize-containerd-main-v2` failed tho. I gave it another try but it seems to be out of the scope of my PR. 

Remaining at your disposal.

Best,

**Comment by bart0sh** [comment]
@TommyStarK Thank you for the PR. Please, ignore `pull-kubernetes-e2e-inplace-pod-resize-containerd-main-v2` failure. It's not related to your changes.

**Comment by TommyStarK** [comment]
Thanks for confirming @bart0sh. Please don't hesitate if there is any change I have to make. 

Best,

**Comment by TommyStarK** [comment]
I do have one question @bart0sh,
I have other open PRs but I wonder why the job `pull-kubernetes-e2e-inplace-pod-resize-containerd-main-v2` is only being triggered here ? I don't even see it in the other check lists. Is it because the changes affect the kubelet's tests ? Sorry for the dummy question, still trying to "find my way" into this project 😅

**Comment by aojea** [comment]
> pull-kubernetes-e2e-inplace-pod-resize-containerd-main-v2

check the job configuration in kubernetes/test-infra

image

there are some docs in that repo that explain the different prow jobs options

**Comment by TommyStarK** [comment]
Hello @aojea, @bart0sh 

I took a  look at `kubernetes/test-infra` and git blame the `config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml` file.
Not really sure about what I should look at, and I definitely don't understand everything. Maybe those commits could have potentially affected the job behaviour:
- https://github.com/kubernetes/test-infra/commit/c420e2793042826b40b4d1c9f8f8570f8e93d895
- https://github.com/kubernetes/test-infra/commit/a10b23f693e5ba2287e3bad399759afd8fda640d
- https://github.com/kubernetes/test-infra/commit/8a7c21f0ed48ac2d36e55e77d2b0bf2fe0e1bc3b
Once again I am NOT sure at all I am looking at the correct place. Either way if I am not mistaking this seems out of the scope of my changes. Should I rebase ?

Best,

**Comment by aojea** [comment]
that job is WIP, don't worry about it https://github.com/kubernetes/test-infra/commit/c420e2793042826b40b4d1c9f8f8570f8e93d895

File: pkg/kubelet/kuberuntime/helpers_test.go [hunk 1]
```diff
@@ -233,3 +250,532 @@ func TestGetImageUser(t *testing.T) {
...
+		},
+	}
+
+	for i, test := range tests {
+		seccompProfile := m.getSeccompProfile(test.annotation, test.containerName, test.podSc, test.containerSc, false)
+		assert.Equal(t, test.expectedProfile, seccompProfile, "TestCase[%d]: %s", i, test.description)
+	}
+}
+
+func TestGetSeccompProfileDefaultSeccomp(t *testing.T) {
```
Comment (aojea) [inline pkg/kubelet/kuberuntime/helpers_test.go:685]: this is linux only, why are you moving it to this file?

**Comment by aojea** [comment]
one advice, it is useful if you split independent changes on different commits, it simplifies reviews and avoids to have to revert the everything if is just only one part that caused a problem

File: pkg/kubelet/kuberuntime/helpers_test.go (hunk 1 above)
Comment (TommyStarK) [reply pkg/kubelet/kuberuntime/helpers_test.go]: Sorry about that, I was completely mistaken

**Comment by TommyStarK** [comment]
Something I don't understand @aojea, govet-levee is failing (https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/114373/pull-kubernetes-verify-govet-levee/1604588953866866688#1:build-log.txt%3A82) same error during unit tests but the package path is imported here (https://github.com/kubernetes/kubernetes/blob/42db9970907539b843674ba0012058c10e42b748/pkg/kubelet/kuberuntime/legacy_test.go#L22)

File: pkg/kubelet/kuberuntime/legacy_test.go [hunk 2]
```diff
@@ -56,3 +56,52 @@ func TestLegacyLogSymLink(t *testing.T) {
...
+	containerID := randStringBytes(80)
+	containerName := randStringBytes(70)
+	podName := randStringBytes(128)
+	podNamespace := randStringBytes(10)
+
+	for _, test := range []struct {
+		name       string
+		logSymLink string
+		expected   string
+		shoulError bool
```
Comment (bart0sh) [inline pkg/kubelet/kuberuntime/legacy_test.go:70]: nit: rename shoulError -> shouldError

Comment (TommyStarK) [reply pkg/kubelet/kuberuntime/legacy_test.go]: Thank you :)

**Comment by TommyStarK** [comment]
Hello @bart0sh, @aojea 

If there is anything I can do in order to get an approval please tell me.
Happy holidays :) 

Cheers,

**Comment by TommyStarK** [comment]
Hello @bart0sh,

I cleaned a bit the code to be consistent with the existing tests. Remaining at your disposal if there is any change to make.

Happy new year, best wishes
Cheers

**Comment by csantanapr** [comment]
Hi @TommyStarK @bart0sh , Bug Triage Team for v1.27 here o/
Just wanted to touch base with you for updates on this PR.
Thanks!

**Comment by TommyStarK** [comment]
Hello @csantanapr
Let me know if there is anything I can do

**Comment by bart0sh** [comment]
@csantanapr This PR is pending approval. Assigning approvers.

/assign @mrunalp @derekwaynecarr @dchen1107

**Comment by TommyStarK** [comment]
@SergeyKanzhelev I tried to contact kubelet approvers to move on on those PRs and avoid the lifecycle/stale label

**Comment by derekwaynecarr** [comment]
Thank you for the testing updates!

Apologies for the lag.

/approve
/lgtm

File: pkg/kubelet/kuberuntime/legacy_test.go [hunk 3]
```diff
@@ -56,3 +56,52 @@ func TestLegacyLogSymLink(t *testing.T) {
...
+		{
+			name:        "container ID too short",
+			logSymLink:  filepath.Join(legacyContainerLogsDir, fmt.Sprintf("%s_%s_%s-%s", podName, podNamespace, containerName, containerID[:5])+".log"),
+			expected:    "",
+			shouldError: true,
+		},
+		{
+			name:        "valid path",
+			logSymLink:  filepath.Join(legacyContainerLogsDir, fmt.Sprintf("%s_%s_%s-%s", podName, podNamespace, containerName, containerID)[:251]+".log"),
+			expected:    "zwVGqMZcLVCxaSJlDSYEofkkEYeqkKHqgBpnbPbg",
```
Comment (pacoxu) [inline pkg/kubelet/kuberuntime/legacy_test.go:93]: /hold
the ci failed and my local run also failed for this hardcode.

Comment (pacoxu) [inline pkg/kubelet/kuberuntime/legacy_test.go:93]:
```suggestion
			expected:    containerID[:40],
```

**Comment by TommyStarK** [comment]
@pacoxu sorry for the trouble and the hardcoded value, I've followed your suggestion and I'll be more careful in the future

**Comment by pacoxu** [comment]
No worry. 
Thanks for the quick fix.
/lgtm

**Comment by TommyStarK** [comment]
@pacoxu can we cancel the hold ?

**Comment by pacoxu** [comment]
/ubhold
sorry to forget about it 😄

**Pull request merged**